### PR TITLE
add hook in order to add more permissions checks from external module

### PR DESCRIPTION
**Pull request by altairis-tof**

### CODE REVIEW

File: htdocs/core/lib/security.lib.php [hunk 1]
```diff
@@ -331,10 +331,17 @@ function restrictedArea($user, $features, $objectid=0, $tableandshare='', $featu
...
+			accessforbidden();
     }
 
+	// get more permissions checks from hooks
+	global $hookmanager;
+	$hookmanager->initHooks(array('permissions'));
+	$parameters=array('features'=>$features,'objectid'=>preg_replace("/'/", '', $objectid),'idtype'=>$dbt_select);
+	$reshook=$hookmanager->executeHooks('restricted',$parameters);
+	if ($reshook < 0) accessforbidden();
+
```
Comment (eldy) [inline htdocs/core/lib/security.lib.php]: The standard of hooks is to introduce only "addreplace" hooks that return a non zero value to say replace standard code.
So you must first call the hook and then if reshook is 0 you return the result of standard code checkuseraccestoobject.

Comment (altairis-tof) [reply htdocs/core/lib/security.lib.php]: sorry i do not understand...

**Comment by altairis-tof** [comment]
i now used resarray instead of return value; is it ok ?

File: htdocs/core/lib/security.lib.php (hunk 1 above)
Comment (eldy) [reply htdocs/core/lib/security.lib.php]: The hook is to replace or complete method  "checkUserAccessToObject"
So it must be called before calling checkUserAccessToObject. And the if return of ok is 0, check must also be done by calling checkUserAccessToObject
Returning if permission is access is ok or not can be done into the resarray

**Comment by eldy** [comment]
Nearly.

The hook is to replace or complete method  "checkUserAccessToObject"
So it must be called before calling checkUserAccessToObject. And the if return is 0, check must also be done by calling checkUserAccessToObject (if return is 1, standard called is not done so only checked in hook)
Returning if permission is access is ok or not can be done into the resArray['restricted'] like you did.

**Comment by altairis-tof** [comment]
replaced by #8750 
Please note that it is not replacing or completing checkUserAccessToObject but it is another level of security.
If we do as you said before, it will not work because if the hook returns 0 (not authorized) we surely dont want to call checkUserAccessToObject that may return authorized !

**Pull request closed without merging**